### PR TITLE
chore(history): update to history ^0.14.0 internally

### PR DESCRIPTION
**Pull request by justingreenberg**
original discussion #89 i tried to keep changes as minimal as possible since it sounds like there may be a rewrite happening soon

i can also submit a separate PR for devtools now that it's stable if you want

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -88,7 +88,7 @@ export function syncReduxAndRouter(history, store, selectRouterState = SELECT_ST
   const unsubscribeHistory = history.listen(location => {
     const route = {
       path: createPath(location),
-      state: location.state
+      state: location.state === null ? undefined : location.state
```
Comment (kjbekkelund) [inline src/index.js]: Was this because tests failed?

**Comment by kjbekkelund** [comment]
Hi @justingreenberg, thanks for this! Only had a small question, otherwise I think this is good to go.

It would be great if you created a DevTools PR too!

File: src/index.js (hunk 1 above)
Comment (justingreenberg) [inline src/index.js]: hi @kjbekkelund! yes, the tests were failing due to the way history assigns a `null` value to state, so this was the most lightweight way to keep state `undefined` internally

Comment (kjbekkelund) [inline src/index.js]: Ah, maybe we should change the tests instead? Nice to be consistent with `history`, plus the code is simpler.

**Comment by jlongster** [comment]
:+1:

File: src/index.js (hunk 1 above)
Comment (kjbekkelund) [inline src/index.js]: @justingreenberg If it isn't too much work changing the tests, maybe you could take a look when you've got the opportunity? No hurry — remember to relax and enjoy xmas :)

**Comment by justingreenberg** [comment]
hey guys! @kjbekkelund ok so in :gift: 43be04e i removed the location.state expression and updated tests to expect state to be `null` after interacting with `history`... it's kinda awkward but it makes sense :)

let me know if you get around to reviewing, if not enjoy the holidays! :santa: :christmas_tree:

**Comment by kjbekkelund** [comment]
@justingreenberg Nice, thanks! I'll try to get this tested and merged tomorrow. Great work :100:

**Pull request merged**